File: library/src/main/java/org/benjinus/pdfium/search/TextSearchContext.java
package org.benjinus.pdfium.search;

import android.graphics.RectF;

public interface TextSearchContext {

    void prepareSearch();

    int getPageIndex();

    String getQuery();

    boolean isMatchCase();

    boolean isMatchWholeWord();

    int countResult();

    boolean hasNext();

    boolean hasPrev();

    RectF searchNext();

    RectF searchPrev();

    void startSearch();

    void stopSearch();
}

File: library/src/main/java/org/benjinus/pdfium/search/TextSearchContext.kt
package org.benjinus.pdfium.search

import android.graphics.RectF

interface TextSearchContext {
    fun prepareSearch()
    val pageIndex: Int
    val query: String?
    val isMatchCase: Boolean
    val isMatchWholeWord: Boolean
    fun countResult(): Int
    operator fun hasNext(): Boolean
    fun hasPrev(): Boolean
    fun searchNext(): RectF?
    fun searchPrev(): RectF?
    fun startSearch()
    fun stopSearch()
}